File: src/test/java/io/pivotal/project/ProjectTest.java
package io.pivotal.project;

import org.junit.Test;

import java.time.LocalDate;

import static org.assertj.core.api.Assertions.assertThat;


public class ProjectTest {

    @Test
    public void constructor_createsProjectFromProjectEntity() {
        ProjectEntity projectEntity = new ProjectEntity();
        projectEntity.setId(Integer.MAX_VALUE);
        projectEntity.setName("foo-bar-baz");
        projectEntity.setStartDate(LocalDate.MAX);
        projectEntity.setHourlyRate(Integer.MIN_VALUE);
        projectEntity.setBudget(Integer.MAX_VALUE / 2);

        Project project = new Project(projectEntity);

        assertThat(project.getId()).isEqualTo(projectEntity.getId());
        assertThat(project.getName()).isEqualTo(projectEntity.getName());
        assertThat(project.getStartDate()).isEqualTo(projectEntity.getStartDate());
        assertThat(project.getHourlyRate()).isEqualTo(projectEntity.getHourlyRate());
        assertThat(project.getBudget()).isEqualTo(projectEntity.getBudget());
        assertThat(project.getBurndown()).isEmpty();
    }
}

File: src/test/java/io/pivotal/project/ProjectTest.kt
package io.pivotal.project

import org.junit.Test

import java.time.LocalDate

import org.assertj.core.api.Assertions.assertThat


class ProjectTest {

    //TODO: This dependency is awkward... domain shouldn't depend on persistence
    //TODO: We should probably NOT have this constructor at all...
    @Test
    fun constructor_createsProjectFromProjectEntity() {
        val projectEntity = ProjectEntity(
                id = Integer.MAX_VALUE,
                name = "foo-bar-baz",
                startDate = LocalDate.MAX,
                hourlyRate = Integer.MIN_VALUE,
                budget = Integer.MAX_VALUE / 2
        )

        val project = Project(projectEntity)

        assertThat(project.id).isEqualTo(projectEntity.id)
        assertThat(project.name).isEqualTo(projectEntity.name)
        assertThat(project.startDate).isEqualTo(projectEntity.startDate)
        assertThat(project.hourlyRate).isEqualTo(projectEntity.hourlyRate)
        assertThat(project.budget).isEqualTo(projectEntity.budget)
        assertThat(project.burndown).isEmpty()
    }
}